src/hooks/brla/useBRLAKYCProcess/index.tsx

```diff
@@ -0,0 +1,141 @@
+import { useCallback, useState, useEffect } from 'react';
+import { useQueryClient } from '@tanstack/react-query';
+
+import { KYCStatus } from '../../../components/BrlaComponents/VerificationStatus';
+import { useOfframpActions } from '../../../stores/offrampStore';
+import { useOfframpSubmission } from '../useOfframpSubmission';
+import { useKYCStatusQuery } from '../useKYCStatusQuery';
+import { KYCFormData } from '../useKYCForm';
+
+export interface BrlaKycStatus {
+  status: string;
+}
+
+enum KYCResponseStatus {
+  PENDING = 'PENDING',
+  SUCCESS = 'SUCCESS',
+  FAILED = 'FAILED',
+}
+
+const STATUS_MESSAGES = {
+  PENDING: 'Estamos verificando seus dados, aguarde',
+  SUCCESS: 'Você foi validado',
+  FAILED: 'Seu KYC foi rejeitado',
+  ERROR: 'Erro durante a verificação',
+};
+
+type StatusMessageType = (typeof STATUS_MESSAGES)[keyof typeof STATUS_MESSAGES];
+
+const ERROR_DISPLAY_DURATION_MS = 3000;
+const SUCCESS_DISPLAY_DURATION_MS = 2000;
+
+const delay = (ms: number): Promise<void> => new Promise((resolve) => setTimeout(resolve, ms));
+
+const useVerificationStatusUI = () => {
+  const [verificationStatus, setVerificationStatus] = useState<KYCStatus>(KYCStatus.PENDING);
+  const [statusMessage, setStatusMessage] = useState<StatusMessageType>(STATUS_MESSAGES.PENDING);
+
+  const updateStatus = useCallback((status: KYCStatus, message: StatusMessageType) => {
+    setVerificationStatus(status);
+    setStatusMessage(message);
+  }, []);
+
+  return {
+    verificationStatus,
+    statusMessage,
+    updateStatus,
+    resetToDefault: useCallback(() => {
+      setVerificationStatus(KYCStatus.PENDING);
+      setStatusMessage(STATUS_MESSAGES.PENDING);
+    }, []),
+  };
+};
+
+export function useKYCProcess(setIsOfframpSummaryDialogVisible: (isVisible: boolean) => void) {
+  const { verificationStatus, statusMessage, updateStatus, resetToDefault } = useVerificationStatusUI();
+  const [isSubmitted, setIsSubmitted] = useState(false);
+
+  const [cpf, setCpf] = useState<string | null>(null);
+
+  const queryClient = useQueryClient();
+  const { data: kycResponse, error } = useKYCStatusQuery(cpf);
+  const { setOfframpKycStarted, resetOfframpState } = useOfframpActions();
+
+  const handleBackClick = useCallback(() => {
+    setOfframpKycStarted(false);
+    resetOfframpState();
+  }, [setOfframpKycStarted, resetOfframpState]);
+
+  const handleError = useCallback(
+    (errorMessage?: string) => {
+      console.error(errorMessage || 'KYC process error');
+      updateStatus(KYCStatus.FAILED, STATUS_MESSAGES.ERROR);
+
+      return delay(ERROR_DISPLAY_DURATION_MS).then(() => {
+        resetToDefault();
+        handleBackClick();
+      });
+    },
+    [handleBackClick, updateStatus, resetToDefault],
+  );
+
+  const proceedWithOfframp = useOfframpSubmission(handleError, setIsOfframpSummaryDialogVisible);
+
+  const handleFormSubmit = useCallback(
```

Comment:
This function is missing a call to the `createSubaccount()` endpoint of our signer service with the relevant fields. Only then, can we query the KYC status of the user and expect to get a meaningful response.